pabc.client/src/plugins/routerGuards.ts

```diff
@@ -0,0 +1,111 @@
+import type { App } from "vue";
+import type { Router } from "vue-router";
+import type { RouteLocationNormalized, NavigationGuardNext } from "vue-router";
+import { useAuthStore } from "@/stores/auth";
+
+function authGuard(
+  to: RouteLocationNormalized,
+  from: RouteLocationNormalized,
+  next: NavigationGuardNext
+) {
+  const authStore = useAuthStore();
+  const requiresAuth = to.matched.some((record) => record.meta.requiresAuth);
+
+  if (requiresAuth && !authStore.isAuthenticated) {
+    // Don't redirect if we're likely in the middle of authentication callback
+    // Give the backend time to establish the session
+    if (authStore.isLoading) {
+      // Still initializing, allow navigation to continue
+      return next();
+    }
+
+    return next({
```

Comment:
hier moet je net als in de fetch wrapper naar de login gestuurd worden (location.href)